components/webln.js

```diff
@@ -0,0 +1,106 @@
+import { createContext, createRef, useContext, useEffect, useImperativeHandle, useState } from 'react'
+
+const WebLNContext = createContext({})
+export const WebLNContextRef = createRef()
+
+const fetchWebLNProvider = async () => {
+  // sync provider from local storage
+  // TODO check if there is a better, more secure option
+  const configJson = window.localStorage.getItem('bc:config')
+  if (configJson) {
+    const WebLNProviders = await import('@getalby/bitcoin-connect').then((mod) => mod.WebLNProviders)
+    const config = JSON.parse(configJson)
+    if (config.connectorType === 'lnbits') return new WebLNProviders.LnbitsWebLNProvider(config.lnbitsInstanceUrl, config.lnbitsAdminKey)
```

Comment:
why does stacker news need to care about the provider type? this is ideally handled by Bitcoin Connect. Stacker news should just use WebLN.

Reply:
I agree, we shouldn't need to care. See [this comment](https://github.com/stackernews/stacker.news/pull/715#discussion_r1437560662).